Tools/TestWebKitAPI/Tests/WebKitCocoa/UseSystemAppearance.mm

```diff
@@ -0,0 +1,104 @@
+{
+    TestWebKitAPI::Util::run(&_doneLoading);
+}
+
+@end
+
+TEST(UseSystemAppearance, UserStyleSheetParsing)
+{
+    RetainPtr styleSheet1 = adoptNS([[_WKUserStyleSheet alloc] initWithSource:@"@media (prefers-dark-interface) { #test { color: green } }" forMainFrameOnly:NO]);
+    RetainPtr styleSheet2 = adoptNS([[_WKUserStyleSheet alloc] initWithSource:@"@media not screen and (prefers-dark-interface) { #test { color: green } }" forMainFrameOnly:NO]);
+
+    RetainPtr configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    [[configuration userContentController] _addUserStyleSheet:styleSheet1.get()];
+    [[configuration userContentController] _addUserStyleSheet:styleSheet2.get()];
+
+    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 400, 300) configuration:configuration.get()]);
+    [webView synchronouslyLoadHTMLString:@"<div id=test></div>"];
+
+    EXPECT_WK_STREQ("rgb(0, 0, 0)", [webView stringByEvaluatingJavaScript:@"getComputedStyle(test).color"]);
+
+    [webView _setUseSystemAppearance:YES];
+    EXPECT_WK_STREQ("rgb(0, 128, 0)", [webView stringByEvaluatingJavaScript:@"getComputedStyle(test).color"]);
+
+    [webView _setUseSystemAppearance:NO];
+    EXPECT_WK_STREQ("rgb(0, 0, 0)", [webView stringByEvaluatingJavaScript:@"getComputedStyle(test).color"]);
+}
+
+TEST(UseSystemAppearance, UserStyleSheetParsingWebKitLegacy)
+{
+    RetainPtr webView = adoptNS([[WebView alloc] initWithFrame:NSMakeRect(0, 0, 400, 300) frameName:@"" groupName:@"UserStyleSheetParsingWebKitLegacyTest"]);
+
+    RetainPtr delegate = adoptNS([UserStyleSheetParsingWebKitLegacyTest new]);
+    [webView setFrameLoadDelegate:delegate.get()];
+
+    [[webView mainFrame] loadHTMLString:@"<div id=test></div>" baseURL:nil];
+    [delegate waitForLoadToFinish];
+
+    [WebView _addUserStyleSheetToGroup:@"UserStyleSheetParsingWebKitLegacyTest" world:[WebScriptWorld world] source:@"@media (prefers-dark-interface) { #test { color: green } }" url:nil includeMatchPatternStrings:nil excludeMatchPatternStrings:nil injectedFrames:WebInjectInAllFrames];
+    [WebView _addUserStyleSheetToGroup:@"UserStyleSheetParsingWebKitLegacyTest" world:[WebScriptWorld world] source:@"@media not screen and (prefers-dark-interface) { #test { color: green } }" url:nil includeMatchPatternStrings:nil excludeMatchPatternStrings:nil injectedFrames:WebInjectInAllFrames];
+
+    EXPECT_WK_STREQ("rgb(0, 0, 0)", [webView stringByEvaluatingJavaScriptFromString:@"getComputedStyle(test).color"]);
+
+    [webView _setUseSystemAppearance:YES];
+    EXPECT_WK_STREQ("rgb(0, 128, 0)", [webView stringByEvaluatingJavaScriptFromString:@"getComputedStyle(test).color"]);
+
+    [webView _setUseSystemAppearance:NO];
+    EXPECT_WK_STREQ("rgb(0, 0, 0)", [webView stringByEvaluatingJavaScriptFromString:@"getComputedStyle(test).color"]);
+
+    [WebView _removeUserStyleSheetsFromGroup:@"UserStyleSheetParsingWebKitLegacyTest" world:[WebScriptWorld world]];
```

Comment:
Is this necessary?

Reply:
Just to avoid unnecessarily leaking state into other tests.